src/viem/saga.ts

```diff
@@ -1,35 +1,238 @@
 import BigNumber from 'bignumber.js'
-import erc20 from 'src/abis/IERC20.json'
-import { tokenAmountInSmallestUnit } from 'src/tokens/saga'
+import erc20 from 'src/abis/IERC20'
+import stableToken from 'src/abis/StableToken'
+import { FeeInfo } from 'src/fees/saga'
+import { encryptComment } from 'src/identity/commentEncryption'
+import { buildSendTx } from 'src/send/saga'
+import { getTokenInfo, tokenAmountInSmallestUnit } from 'src/tokens/saga'
+import { isStablecoin } from 'src/tokens/utils'
+import { chooseTxFeeDetails } from 'src/transactions/send'
+import { TransactionContext } from 'src/transactions/types'
 import Logger from 'src/utils/Logger'
 import { publicClient } from 'src/viem'
+import { ViemWallet } from 'src/viem/getLockableWallet'
 import { getViemWallet } from 'src/web3/contracts'
 import networkConfig from 'src/web3/networkConfig'
 import { call } from 'typed-redux-saga'
-import { getAddress } from 'viem'
-
-// example usage
-// todo(ACT-786): build on this when implementing viem version of sendAndMonitorTransaction
-export function* send(
-  recipientAddress: string,
-  amount: BigNumber,
-  tokenAddress: string,
+import { SimulateContractReturnType, getAddress } from 'viem'
+
+const TAG = 'viem/saga'
+
+/**
+ * Send a payment with viem. The equivalent of buildAndSendPayment in src/send/saga.
+ *
+ * @param options an object containing the arguments
+ * @param options.context the transaction context
+ * @param options.recipientAddress the address to send the payment to
+ * @param options.amount the crypto amount to send
+ * @param options.tokenAddress the crypto token address
+ * @param options.comment the comment on the transaction
+ * @param options.feeInfo an object containing the fee information
+ * @returns
+ */
+export function* sendPayment({
+  context,
+  recipientAddress,
+  amount,
+  tokenAddress,
+  comment,
+  feeInfo,
+}: {
+  context: TransactionContext
+  recipientAddress: string
+  amount: BigNumber
+  tokenAddress: string
   comment: string
-) {
+  feeInfo: FeeInfo
+}) {
   const wallet = yield* call(getViemWallet, networkConfig.viemChain.celo)
-  Logger.debug('tokenAddress', tokenAddress, getAddress(tokenAddress))
-  const convertedAmount: string = yield* call(tokenAmountInSmallestUnit, amount, tokenAddress)
-  Logger.debug('convertedAmount', convertedAmount)
-
-  const { request } = yield* call([publicClient.celo, 'simulateContract'], {
-    address: getAddress(tokenAddress),
-    abi: erc20.abi,
-    functionName: 'transfer',
-    account: wallet.account,
-    args: [recipientAddress, convertedAmount],
+
+  Logger.debug(
+    TAG,
+    'Transferring token',
+    context.description ?? 'No description',
+    context.id,
+    tokenAddress,
+    amount,
+    feeInfo
+  )
+
+  // TODO: add standby transaction
```

Comment:
when's this going to get done?

Reply:
if it's part of a Linear issue can you reference it here? If it's not, but you personally plan to do it, can you add your name or username to the todo?

Reply:
I mentioned this in the PR description, its part of the same linear issue and there will be a followup PR. I can add my name to the TODO if you feel strongly about this.